pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +522,24 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasAppLabel(result *newcmd.AppResult) (bool, error) {
```

Comment:
@bparees not sure that this is enough... when we set the app label, we'll set it in other places like the template inside a deployment config, which we aren't testing here.

Maybe we can add a SetLabelsWithMode function in utils/labels.go that allows us to specify a mode and allow  us to test for conflicts. And actually we don't just want to test for the label being present/absent. If it has the same value as what we're trying to set, then I don't see any harm in setting it.